mlir/include/mlir/Target/LLVMIR/ModuleImport.h

```diff
@@ -362,9 +362,12 @@ class ModuleImport {
   /// Converts the callee's function type. For direct calls, it converts the
   /// actual function type, which may differ from the called operand type in
   /// variadic functions. For indirect calls, it converts the function type
-  /// associated with the call instruction. Returns failure when the call and
-  /// the callee are not compatible or when nested type conversions failed.
-  FailureOr<LLVMFunctionType> convertFunctionType(llvm::CallBase *callInst);
+  /// associated with the call instruction. When the call and the callee are not
+  /// compatible (or when nested type conversions failed), emit a warning but
+  /// attempt translation using a bitcast and an indirect call (in order
+  /// represent valid and verified LLVM IR).
+  FailureOr<LLVMFunctionType> convertFunctionType(llvm::CallBase *callInst,
+                                                  Value &castResult);
```

Comment:
nit: I wonder if there would be a better name fro castResult? Isn't something like calleeAddress, calleeOperandToAdd, or extraCalleeOperand better? Maybe also briefly explain in the comment that the extra output argument of the function should be used to build the indirect call.

Reply:
Good point, rewrote a good part of the comments and renamed it to `indirectCallVal`.

Reply:
Nice, ready to go I would say.